tools/content-generator/src/main/java/org/projectnessie/tools/contentgenerator/cli/GenerateContent.java

```diff
@@ -130,98 +128,102 @@ public void execute() throws BaseNessieClientServerException {
                         Integer.toString(i)))
             .collect(Collectors.toList());
 
-    Branch defaultBranch;
-    if (defaultBranchName == null) {
-      // Use the server's default branch.
-      defaultBranch = api.getDefaultBranch();
-    } else {
-      // Use the specified default branch.
-      try {
-        defaultBranch = (Branch) api.getReference().refName(defaultBranchName).get();
-      } catch (NessieReferenceNotFoundException e) {
-        // Create branch if it does not exist.
+    try (NessieApiV1 api = createNessieApiInstance()) {
+      Branch defaultBranch;
+      if (defaultBranchName == null) {
+        // Use the server's default branch.
         defaultBranch = api.getDefaultBranch();
-        defaultBranch =
-            (Branch)
-                api.createReference()
-                    .reference(Branch.of(defaultBranchName, defaultBranch.getHash()))
-                    .sourceRefName(defaultBranch.getName())
-                    .create();
+      } else {
+        // Use the specified default branch.
+        try {
+          defaultBranch = (Branch) api.getReference().refName(defaultBranchName).get();
+        } catch (NessieReferenceNotFoundException e) {
+          // Create branch if it does not exist.
```

Comment:
is it safe to create a branch automatically? If yes, then I think we should document that.

Reply:
this code hasn't been touched by this PR